### ISSUE TITLE
Make Publish actually publish

### ISSUE THREAD

**Issue by joejulian** [samsung-cnct/k2#655]
The Publish stage of the Jenkins build doesn't actually publish. The environment variables we've tried testing against are null during the master build (but not during the pr build). This prevents master builds from ever successfully pushing to quay.

**Comment by joejulian** [comment]
These are the `env.*` available during the master build.
```
Name: BRANCH_NAME -> Value master
Name: BUILD_DISPLAY_NAME -> Value #11
Name: BUILD_ID -> Value 11
Name: BUILD_NUMBER -> Value 11
Name: BUILD_TAG -> Value jenkins-k2-jj-master-11
Name: BUILD_URL -> Value https://common-jenkins.kubeme.io/job/k2-jj/job/master/11/
Name: CLASSPATH -> Value 
Name: HUDSON_HOME -> Value /var/jenkins_home
Name: HUDSON_SERVER_COOKIE -> Value 6007e83f66d1eaa8
Name: HUDSON_URL -> Value https://common-jenkins.kubeme.io/
Name: JENKINS_HOME -> Value /var/jenkins_home
Name: JENKINS_SERVER_COOKIE -> Value 6007e83f66d1eaa8
Name: JENKINS_URL -> Value https://common-jenkins.kubeme.io/
Name: JOB_BASE_NAME -> Value master
Name: JOB_DISPLAY_URL -> Value https://common-jenkins.kubeme.io/job/k2-jj/job/master/display/redirect
Name: JOB_NAME -> Value k2-jj/master
Name: JOB_URL -> Value https://common-jenkins.kubeme.io/job/k2-jj/job/master/
Name: RUN_CHANGES_DISPLAY_URL -> Value https://common-jenkins.kubeme.io/job/k2-jj/job/master/11/display/redirect?page=changes
Name: RUN_DISPLAY_URL -> Value https://common-jenkins.kubeme.io/job/k2-jj/job/master/11/display/redirect
```

**Comment by joejulian** [comment]
Still didn't work even though the correct strings were being tested against.

**Comment by joejulian** [comment]
Fixed in PR #656
